### ISSUE TITLE
Default values for parameters used in if change the compiled model

### ISSUE THREAD

**Issue by cossartque** [dynawo/dynawo#2906]
If we have this model:
<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
parameter Boolean MyBooleanParameter = true "Default value at true";
Real MyVariable;

equation
if MyBooleanParameter then
  MyVariable = 1;
else
  MyVariable = 3;
end if;
<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<

Then we can change MyBooleanParameter in the .par, no problem, the parameter exists and can be modified, BUT the equation has been simplified when compiling using the default value and we will always have MyVariable = 1;

And of course there is no warning as the parameter exists, it is just not used as the if condition has disappeared.

**Comment by rosiereflo** [comment]
I don't think we can do anything about this one as the simplification is done on OpenModelica side. There is no information left about the original if in the generated code.
We need to avoid giving default values to boolean parameters.
